providers/azure_test.go

```diff
@@ -122,82 +122,134 @@ func testAzureBackend(payload string) *httptest.Server {
 	defer b.Close()
 
 	bURL, _ := url.Parse(b.URL)
 	p := testAzureProvider(bURL.Host)
 
 	session := CreateAuthorizedSession()
-	email, err := p.GetEmailAddress(context.Background(), session)
+	err := p.EnrichSessionState(context.Background(), session)
 	assert.Equal(t, nil, err)
-	assert.Equal(t, "user@windows.net", email)
+	assert.Equal(t, "user@windows.net", session.Email)
 }
 
-func TestAzureProviderGetEmailAddressGetUserPrincipalName(t *testing.T) {
+func TestAzureProviderEnrichSessionStateGetUserPrincipalName(t *testing.T) {
 	b := testAzureBackend(`{ "mail": null, "otherMails": [], "userPrincipalName": "user@windows.net" }`)
 	defer b.Close()
 
 	bURL, _ := url.Parse(b.URL)
 	p := testAzureProvider(bURL.Host)
 
 	session := CreateAuthorizedSession()
-	email, err := p.GetEmailAddress(context.Background(), session)
+	err := p.EnrichSessionState(context.Background(), session)
 	assert.Equal(t, nil, err)
-	assert.Equal(t, "user@windows.net", email)
+	assert.Equal(t, "user@windows.net", session.Email)
 }
 
-func TestAzureProviderGetEmailAddressFailToGetEmailAddress(t *testing.T) {
+func TestAzureProviderEnrichSessionStateFailToEnrichSessionState(t *testing.T) {
 	b := testAzureBackend(`{ "mail": null, "otherMails": [], "userPrincipalName": null }`)
 	defer b.Close()
 
 	bURL, _ := url.Parse(b.URL)
 	p := testAzureProvider(bURL.Host)
 
 	session := CreateAuthorizedSession()
-	email, err := p.GetEmailAddress(context.Background(), session)
+	err := p.EnrichSessionState(context.Background(), session)
 	assert.Equal(t, "type assertion to string failed", err.Error())
-	assert.Equal(t, "", email)
+	assert.Equal(t, "", session.Email)
 }
 
-func TestAzureProviderGetEmailAddressEmptyUserPrincipalName(t *testing.T) {
+func TestAzureProviderEnrichSessionStateEmptyUserPrincipalName(t *testing.T) {
 	b := testAzureBackend(`{ "mail": null, "otherMails": [], "userPrincipalName": "" }`)
 	defer b.Close()
 
 	bURL, _ := url.Parse(b.URL)
 	p := testAzureProvider(bURL.Host)
 
 	session := CreateAuthorizedSession()
-	email, err := p.GetEmailAddress(context.Background(), session)
+	err := p.EnrichSessionState(context.Background(), session)
 	assert.Equal(t, nil, err)
-	assert.Equal(t, "", email)
+	assert.Equal(t, "", session.Email)
 }
 
-func TestAzureProviderGetEmailAddressIncorrectOtherMails(t *testing.T) {
+func TestAzureProviderEnrichSessionStateIncorrectOtherMails(t *testing.T) {
 	b := testAzureBackend(`{ "mail": null, "otherMails": "", "userPrincipalName": null }`)
 	defer b.Close()
 
 	bURL, _ := url.Parse(b.URL)
 	p := testAzureProvider(bURL.Host)
 
 	session := CreateAuthorizedSession()
-	email, err := p.GetEmailAddress(context.Background(), session)
+	err := p.EnrichSessionState(context.Background(), session)
 	assert.Equal(t, "type assertion to string failed", err.Error())
-	assert.Equal(t, "", email)
+	assert.Equal(t, "", session.Email)
+}
+
+func TestAzureProviderEnrichSessionStateFromIDToken(t *testing.T) {
```

Comment:
Can you refactor these to use test tables? While this style tests are still in the code base, in general we don't allow new merges using it so we can ween ourselves off of this style for better, cleaner tests in the long term.

Test tables are OK. Ginkgo is our preferred testing format, but I don't think the refactor to bring Ginkgo test suit support to the `providers` package has merged yet.

Reply:
Here's the PR that includes the Ginkgo addition and some sample tests: https://github.com/oauth2-proxy/oauth2-proxy/pull/630/files#diff-233c3f5da886b2facdebcec5f9802b1df67ef0252d55be63a882a6eb49661d85R227

This PR is almost at the finish line if you want to build off its work with Ginkgo tests.

Reply:
done